src/clients/Deluge.ts

```diff
@@ -0,0 +1,266 @@
+import { InjectionResult } from "../constants.js";
+import { CrossSeedError } from "../errors.js";
+import { Label, logger } from "../logger.js";
+import { Metafile } from "../parseTorrent.js";
+import { getRuntimeConfig } from "../runtimeConfig.js";
+import { Searchee } from "../searchee.js";
+import { TorrentClient } from "./TorrentClient.js";
+import { extractCredentialsFromUrl } from "../utils.js";
+import fetch, { Headers, Response } from "node-fetch";
+
+interface DelugeResponse {
+	error?: {
+		message?: string;
+		code?: DelugeErrorCode;
+	};
+	result?: string | boolean;
+}
+enum DelugeErrorCode {
+	NO_AUTH = 1,
+	BAD_METHOD = 2,
+	CALL_ERR = 3,
+	RPC_FAIL = 4,
+	BAD_JSON = 5,
+}
+interface TorrentInfo {
+	complete: boolean;
+	save_path?: string;
+}
+
+export default class Deluge implements TorrentClient {
+	private delugeCookie = null;
+	private delugeLabel = "cross-seed";
+	private isLabelEnabled: boolean;
+
+	/**
+	 * validates the login and host for deluge webui
+	 */
+	async validateConfig(): Promise<void> {
+		try {
+			await this.authenticate();
+		} catch (validateError) {
+			throw new CrossSeedError(validateError);
+		}
+		this.isLabelEnabled = await this.labelEnabled();
+	}
+
+	/**
+	 * connects and authenticates to the webui
+	 */
+	private async authenticate(): Promise<void> {
+		const { delugeRpcUrl } = getRuntimeConfig();
+		const { href, password } = extractCredentialsFromUrl(
+			delugeRpcUrl
+		).unwrapOrThrow(new Error("delugeRpcUrl must be percent-encoded"));
```

Comment:
sorry, I think I gave some bad advice in the last review. I think you were right originally with `CrossSeedError` on line 54, and then validateConfig doesn't need to wrap it in try/catch at all.